core/src/main/java/com/yahoo/imapnio/async/data/QResyncSeqMatchData.java

```diff
@@ -0,0 +1,40 @@
+package com.yahoo.imapnio.async.data;
+
+import javax.annotation.Nonnull;
+
+/**
+ * This class models the QRESYNC Sequence match data defined in https://tools.ietf.org/html/rfc7162#page-26.
+ */
+public class QResyncSeqMatchData {
+    /** Message sequence number. */
+    private MessageNumberSet[] knownSequenceSet;
+
+    /** Corresponding UIDs. */
+    private MessageNumberSet[] knownUidSet;
+
+    /**
+     * Constructor.
+     * @param knownSequenceSet message sequence numbers
+     * @param knownUidSet UIDs
+     */
+    public QResyncSeqMatchData(@Nonnull final MessageNumberSet[] knownSequenceSet, @Nonnull final MessageNumberSet[] knownUidSet) {
+        this.knownSequenceSet = knownSequenceSet;
```

Comment:
Could u help me put @Nonnull in these 2 member variables too?

Reply:
done